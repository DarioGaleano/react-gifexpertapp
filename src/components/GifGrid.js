import React from 'react';
// import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
	console.log({ category });
	const { data, loading } = useFetchGifs(category);

	return (
		<>
			<h3 className='animate__animated animate__fadeIn'>{category}</h3>
			{loading && <p className='animate__animated animate__flash'>Loading...</p>}
			<div className='card-grid'>
				{data.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</>
	);
};

GifGrid.propTypes = {};
