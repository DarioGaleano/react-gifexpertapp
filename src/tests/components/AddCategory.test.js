import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { AddCategory } from '../../components/AddCategory';
describe('Tests on <AddCategory />', () => {
	const setCategory = jest.fn();
	let wrapper = shallow(<AddCategory setCategory={setCategory} />);

	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategory={setCategory} />);
	});

	test('should show correctyle ', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should change the textbox', () => {
		const input = wrapper.find('input');
		const value = 'Hola Mundo';
		input.simulate('change', { target: { value } });
	});

	test('dont should to post the info with submit', () => {
		wrapper.find('form').simulate('submit', { preventDefault() {} });
		expect(setCategory).not.toHaveBeenCalled();
	});

	test('should call setCategory and clean the textbox', () => {
		const value = 'Hello World';
		wrapper.find('input').simulate('change', { target: { value } });
		wrapper.find('form').simulate('submit', { preventDefault() {} });

		expect(setCategory).toHaveBeenCalled();
		expect(setCategory).toHaveBeenCalledWith(expect.any(Function));
		expect(wrapper.find('input').prop('value')).toBe('');
	});
});
