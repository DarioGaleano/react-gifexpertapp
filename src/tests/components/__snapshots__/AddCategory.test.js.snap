// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory /> should show correctyle  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
